lib/services/watcher.go

```diff
@@ -616,8 +616,8 @@ func (p *lockCollector) processEventAndUpdateCurrent(ctx context.Context, event
 // notifyStale is called when the maximum acceptable staleness (if specified)
 // is exceeded.
 func (p *lockCollector) notifyStale() {
-	p.fanout.Emit(types.Event{Type: types.OpUnreliable})
 	p.currentRW.Lock()
+	p.fanout.Emit(types.Event{Type: types.OpUnreliable})
```

Comment:
It's weird to me to see code between the lock and the deferred unlock.

I would either:

```
lock
emit
unlock (if this is valid)
```

or 

```
lock
defer unlock
emit
```

Reply:
updated, thanks!